### ISSUE TITLE
ValueError when reading a OME-TIFF file

### ISSUE THREAD

**Issue by etiennebergeron** [filippocastelli/pyometiff#1]
Hello,
I'm trying to read 2D OME-TIFF sample files that I found on the OME website (https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/tubhiswt-2D/) with the package _pyometiff_. Here is my code:
```
import pyometiff
reader = pyometiff.OMETIFFReader('tubhiswt_C0.ome.tif')
img_array, metadata, xml_metadata = reader.read()
```
When I run it, I get this error:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-10-bd3614b76a8e> in <module>
      1 reader = pyometiff.OMETIFFReader('tubhiswt_C0.ome.tif')
----> 2 img_array, metadata, xml_metadata = reader.read()

~\miniconda3\envs\tpGBM\lib\site-packages\pyometiff\omereader.py in read(self)
     37     def read(self) -> (np.ndarray, dict, str):
     38         self.array, self.omexml_string = self._open_tiff(self.fpath)
---> 39         self.metadata = self.parse_metadata(self.omexml_string)
     40         return self.array, self.metadata, self.omexml_string
     41 

~\miniconda3\envs\tpGBM\lib\site-packages\pyometiff\omereader.py in parse_metadata(self, omexml_string)
    131                 self.imageseries
    132             ).Detector.get_ID()
--> 133             metadata["DetectorType"] = self.ox.instrument(
    134                 self.imageseries
    135             ).Detector.get_Type()

~\miniconda3\envs\tpGBM\lib\site-packages\pyometiff\omexml.py in get_Type(self)
   1172 
   1173         def get_Type(self):
-> 1174             return get_int_attr(self.node, "Type")
   1175         def set_Type(self, value):
   1176             self.node.set("Type", str(value))

~\miniconda3\envs\tpGBM\lib\site-packages\pyometiff\omexml.py in get_int_attr(node, attribute)
    294     '''Cast an element attribute to an int or return None if not present'''
    295     attr = node.get(attribute)
--> 296     return None if attr is None else int(attr)
    297 
    298 def make_text_node(parent, namespace, tag_name, text):

ValueError: invalid literal for int() with base 10: 'PMT'
```
I also tried with another OME-TIFF file from https://docs.openmicroscopy.org/ome-model/5.6.3/ome-tiff/data.html, for example **00001_01.ome.tiff**, and the reading works just fine.

Do you know why the first ones can't be read?

Thanks for your help,

Etienne

**Comment by filippocastelli** [comment]
Hey! thanks for the feedback, I'll look into both issues right away and keep you updated.

**Comment by filippocastelli** [comment]
The error was caused by an incorrect parsing of the `Detector.Type` OMEXML field, which I seem to have parsed as an integer when it really was a string.

https://www.openmicroscopy.org/Schemas/Documentation/Generated/OME-2016-06/ome_xsd.html#Detector

        def get_Type(self):
            # return get_int_attr(self.node, "Type")  <<< this was causing the error
            return self.node.get("Type")
        def set_Type(self, value):
            self.node.set("Type", str(value))
        Type = property(get_Type, set_Type)

by parsing it as a string the above code works as expected but the `tifffile` dependency still raises a warning:
`OME series: failed to read 'tubhiswt_C1.ome.tif'` 

This is solved in commit ba7ce60c78b654380a54fe5da493b51686d1451b

I'll draft a new release after fixing the other issue you opened.

**Comment by etiennebergeron** [comment]
Perfect, thank you for the fast correction!

**Comment by filippocastelli** [comment]
Now included in v0.0.6 https://github.com/filippocastelli/pyometiff/tree/0.0.6
